### ISSUE TITLE
adam-cli POM invalid: maven.build.timestamp

### ISSUE THREAD

**Issue by ryan-williams** [bigdatagenomics/adam#1004]
Guacamole (which depends on ADAM) is showing an error parsing the adam-cli POM; here is debug output from running `mvn -X package` there:

```
[WARNING] Invalid POM for org.bdgenomics.adam:adam-cli_2.10:jar:0.19.0, transitive dependencies (if any) will not be available, enable debug logging for more details: Some problems were encountered while processing the POMs:
[ERROR] Resolving expression: '${maven.build.timestamp}': Detected the following recursive expression cycle in 'maven.build.timestamp': [maven.build.timestamp] @
```

[This line](https://github.com/bigdatagenomics/adam/blob/f8c3133ae75947fde0a29d63f025f81fbcdb58f9/adam-cli/pom.xml#L15) seems to be the culprit.

Anyone have context on why it's there / what it's supposed to be doing?

**Comment by fnothaft** [comment]
CC @heuermh

**Comment by heuermh** [comment]
That expression is used in the About class here
https://github.com/bigdatagenomics/adam/blob/f8c3133ae75947fde0a29d63f025f81fbcdb58f9/adam-cli/src/main/java-templates/org/bdgenomics/adam/cli/About.java#L25

It is repeated in the pom file as a workaround for a bug with the maven assembly plugin (I believe, would have to dig for context).  I've never seen this maven error before, what version are you using?

**Comment by fnothaft** [comment]
As an aside, why does Guacamole depend on the `adam-cli` artifact? I would expect it to only need `adam-core`?

**Comment by heuermh** [comment]
Ouch, found it, issue open since 2009.
https://issues.apache.org/jira/browse/MRESOURCES-99

There is a workaround but I found it to be unreliable (on a different project).

**Comment by heuermh** [comment]
> As an aside, why does Guacamole depend on the ```adam-cli``` artifact? I would expect it to only need ```adam-core```?

All of my downstream repos do, and I don't believe I've ever seen this error

https://github.com/heuermh/adam-commands
https://github.com/heuermh/adam-examples
https://github.com/heuermh/adam-plugins

**Comment by ryan-williams** [comment]
> I've never seen this maven error before, what version are you using?

```
$ mvn --version
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T16:41:47+00:00)
```

**Comment by ryan-williams** [comment]
> As an aside, why does Guacamole depend on the adam-cli artifact? I would expect it to only need adam-core?

Good point, seems like we don't need it :) will remove it, but maybe still good to understand why the error was happening.

**Comment by fnothaft** [comment]
> but maybe still good to understand why the error was happening.

Agreed.

**Comment by ryan-williams** [comment]
@heuermh I just cloned adam-commands, ran `mvn package`, and saw the warning:

```
[WARNING] Invalid POM for org.bdgenomics.adam:adam-cli_2.10:jar:0.19.1-SNAPSHOT, transitive dependencies (if any) will not be available, enable debug logging for more details
```

**Comment by heuermh** [comment]
Yeah just saw that, with ```mvn --debug ...```
```bash
[DEBUG] Skipped remote request for org.bdgenomics.adam:adam-cli_2.10:0.19.1-SNAPSHOT/maven-metadata.xml, locally installed metadata up-to-date.
[WARNING] The POM for org.bdgenomics.adam:adam-cli_2.10:jar:0.19.1-SNAPSHOT is invalid, transitive dependencies (if any) will not be available: 1 problem was encountered while building the effective model for org.bdgenomics.adam:adam-cli_2.10:0.19.1-SNAPSHOT
[ERROR] Resolving expression: '${maven.build.timestamp}': Detected the following recursive expression cycle in 'maven.build.timestamp': [maven.build.timestamp] @
```

**Comment by heuermh** [comment]
Fixed by #1005
